### DISCUSSION TITLE
Return old focus system and remove stamina

### DISCUSSION THREAD

**Discussion by Seire** [cataclysmbn/Cataclysm-BN#2041] [General]
# Morale
As far as I understood, BN is not only about returning sci-fi into the game, but is also about improving this grindy and slow gameplay of DDA. I believe, one of the biggest problems is a slow pace of progression. Leveling takes too long. players loses too much to deaths. Everyone is supper careful or savescumming, almost no more YASD. 
Speeding up leveling of a character seems like a most obvious decision. So, how about returning the old cataclysm focus system without heavy penalties from using multiple morale boosts? This will give us better focus regen and bigger focus pull. And this leads to faster progression, less grind, and therefore more fun.
Honestly, at my current session I want to try another scenarios, backgrounds and install few BN-ported mods, but I don't wanna loose progress on my current char, since it took efffort and time to reach high skills and tons of loot. For a game with permadeath, this just feels wrong.
Edit: besides, if there will be no penalties, finding more loot will feel more rewarding. This atm there is no real difference between tons of alcohol and foods. However, if every morale source will be unique and independent, this will motivate players to actually care about using different sorts of morale boosters, which will make them useful.

# Stamina
I don't see any reason why it even exists, except DDA realism nonsense. Just why? Makes melee weapons somewhat lacking compared to ranged, feels like a secondary weapon. You don't need ammo and it is silent, but melee depends a lot more on encumbrance values, melee fights tear your equipment and... it has melee range, so, for example, no kiting heavy enemies from afar, animals can outrun you and so on. It already has its own "cons", is there really a need for additional heavy penalty for using melee? Blunt weapons are basically dead and constantly pressing | to wait for stamina regen is just irritating.
Probably will need a nerf for bonuses from skills to make fights harder, tho.

English is not my native language. I believe, in general I am understandable enough, but I apologise for grammar mistakes.

**Comment by Coolthulhu** [comment]
> So, how about returning the old cataclysm focus system without heavy penalties from using multiple morale boosts?

The old system had its own problems:
* Sipping 1 charge from 5 different wines to stack morale bonus
* Eating 1 cookie from 5 different boxes
* Doing a lot of different drugs at once
* Most of the day being spent at low morale, except for that one moment where everything is stacked

Stacking is tedious.
While the new system has its flaws, it at least doesn't favor this tedious stacking.

Morale system could use a rework, but I don't see much benefit from reverting to old one.
I'd rather do something like:
* Extend morale effects when on high morale, so that you don't waste the focus when stacking
* Make focus generate constantly instead of being an equilibrium. Turn it into an "xp pool".

> I don't see any reason why it even exists, except DDA realism nonsense. Just why?

Running away. Before stamina, you could only walk away.
Old DDA required you to either pick Fleet Footed, find a bicycle very fast, or to be very careful not to attract something strong and fast.
Removing stamina and thus running would hurt melee more than keeping stamina.

That said, removing penalties to melee from low stamina is not a bad idea. BN melee doesn't weaken as much at low stamina as DDA melee does.

**Comment by Seire** [comment]
>* Sipping 1 charge from 5 different wines to stack morale bonus
>* Eating 1 cookie from 5 different boxes

Honestly, I don't recall doing that. Maybe we are speaking about different systems, I meant not the oldest Cataclysm focus system, but the older DDA one.
I remember playing DDA, when morale screen didn't have percentages, it had only flat amounts. At those versions morale system was quite close to what we have now. There was amount of focus shifting to a number, defined by used boosts. 
However, regen on high morale was a lot faster and stuff stacked up to 3 times, if I remember correctly, so 3 cookies from the same package, 3 sips of wine etc. Not different packages and bottles, but different types of foods and drinks. So you could get a huge boost to leveling when needed, but in a same time "natural" regen of focus allowed you to level up skills anytime at a normal speed.
Right now it feels like focus barely regenerates. Most of the time in combat I reach 20-30 focus from 80-100 after 5 zeds (or so) and clean up the rest of a group while still being on a 20 focus. Big shift towards max wont happen during playing portable console or reading a chapter of a sci-fi novel.
Maybe simple increase of a regeneration rate will work. Or you can implement a "lowest border of focus", which rises together with equilibrium amount. So if you ve managed to stack equilibrium to 140, your focus wont drop lower than 50-70, so your actions will bring you a lot of exp.

> * Make focus generate constantly instead of being an equilibrium. Turn it into an "xp pool".

Not sure about this one. Too many actions allow you to skip time. Like drying meat in huge amounts, removing/installing "big" parts of vehicles, reading, etc. XP will be too easy to stack. So actions should be less effective in spending exp (so players won't be able to spend it fast) or it will be abusive (read for 3 days, become a master fighter in a single fight). In the first case, what is the reason of having exp pool in a first place? Easy to get, slow to spend.
I do believe, that stacking to get a huge boost when needed is the best reason for this system to exist. You can level normally because of natural equilibrium is not a 0, but if necessary, you can boosts yourself with drugs, drinks and stuff, however, you are not forced to do that every time you need to skill up.

> Running away. Before stamina, you could only walk away.

Ah, right, lol. I have returned to cata just recently (reason Im playing BN, since DDA is way too different from experience I looked for) and never actually used running. Got used to carefully fighting small groups using windows and stuff. 

> removing penalties to melee from low stamina is

Sounds great!

**Comment by Wreckstation** [comment]
an alternate solution to get alleviate grinding that i currently use is to boost skill gain rate by 1.5 (i've found 2.0 is Too fast, but ymmv)

**Comment by Krwak** [comment]
'Make focus generate constantly instead of being an equilibrium. Turn it into an "xp pool".'
I definitely like this one, would save some time and i think it would make it more comfortable to use and also believable.
Changes to focus ought to be faster than they are.

Stamina during fighting and running -
How about halving stamina drain when fighting and leaving running stamina drain as it is?

**Comment by JavelinHeadQueer** [comment]
I think the stamina system could do with a change. Halving it when fighting seems like a good idea. I'm currently running a character with 14 strength and Fast Metabolism to regenerate stamina faster. I run the high strength to lower the stamina drain on two handed weapons that weigh a lot. I think Strength should add to stamina a bit. But if a change like that was made, some weapons like the Ji and Katana would allow for lengthy combat sessions while still having the ability to run away. This could make the early game easier for builds and professions with little in the ways of combat capabilities. But it could also trivialize the early game for characters that are even mildly capable with weapons. I like that the game is punishing. I like that not being careful could mean my character ends up dead. There are professions that can entirely sidestep the early and mid game dangers, like the Bionic Assassin, or Experiment start for Bionic Monster. I think that that diversity is great because the player can decide how _they_ want to play, so professions strong in combat are not a problem to me at all. I think the player should be in control of their own experience. But balancing is also good, and I think the two starts I mentioned are good to have. Right now a friend of mine is playing a computer hacker with the Infected start, meanwhile I'm playing a fast mutant with spears who can't wear shoes. Diversity is good.

I don't want the old morale system to return. On my high morale crafting characters I usually pick Gourmand and Optimist for strong boosts, and pick up an mp3 player and a handheld gaming system to boost it further. I also pick the recipes that take the longest to craft because Focus is rapidly consumed when making say, a cudgel versus a bikini top. Cudgel taking 1 minute and Bikini Top 25 minutes IIRC. With the current way Bright Nights hunger system works, while having Gourmand you eat and drink less than you would in CDDA. So stacking morale is harder and needs more thought. Going from Hungry to Sated on food leaves room for one drink of something, and if you drink more you become nauseous with Gourmand active.

But, I like Coolthulu's idea of having high morale last longer. Maybe a buff to how long the morale boost lasts based on the amount of morale an item or consumable provided? Then again, with Gourmand and Optimist, you can eat a toast-em and drink spiced rum to get something like +51 morale just from those. Add an mp3 player and play some games to pass time _and_ get a morale boost and you can regen 20-30 morale or more in about half an hour to an hour. However, I've noticed that the higher your morale, the harder it is to make it higher, and I usually max out around 110-118, so some modifications could be made. But the problem with learning melee I think is either a problem with how much focus it drains, or a problem with the traits that synergise with melee characters.

The only trait that favors killing is Killer Drive. Killer Drive as it is right now is at best just another small morale boost tacked on to much stronger morale raising synergies and at worst, not picked at all because players find it useless when compared to other morale stacking traits. Gourmand has the exact same cost and you can still use it to stack +30 or more focus even without Optimist adding bonuses to the joy you already get from eating food. Gourmand + Optimist lets you eat a toast-em for +30 morale and drink spiced rum for another +21. I don't think this is a problem. 

I think that the issue is having poor choices when it comes to mood boosts for players who kill a lot of zombies. Killer Drive should receive a buff. It should give you a stronger morale boost for enemies you kill than right now, and I think another morale boost should be added for critting that scales with a percentage of the amount of damage the crit did. This morale boost could be called "Carnage" in the morale window. I also want to suggest a new trait called Adrenaline Junkie to be added to the game. Adrenaline Junkie would have a similar cost to Gourmand and would give a stacking morale bonus for how _many_ enemies you're fighting at the same time, where it makes a check in the code when zombies have detected you, and another check for if when a zombie is within, say, 3 tiles around the character, apply a flat morale buff (or a multiplicative if you're feeling spicy) for each zombie within those tiles that have detected the player. There would also need to be some way for the game to know which zombies have already given the buff so the player can't make the zombies give it to them again when moving in and out of the radius of Adrenaline Junkie. Which could be made with a clause that if they've been in the range of Adrenaline Junkie but not left sight radius of the player, the buff won't be triggered again.

Gamewise the morale boost from crits and kills would make the early game easier to get through on melee characters while boosting their recovery after fights so they can more easily level up crafting. The stacking morale buff from how many enemies are near you would be of use in the midgame when you can take on small groups but are still a bit away from 5 in the relevant melee skills and Heavy Survivor Suit is not available. All these bonuses would be good in the early game for melee builds to recover focus after or while fighting; while the benefits to morale from fighting multiple enemies at the same time would drop off in the late-game as you will have completed most of the gear you want.

Another thing too. To fix morale loss in combat, maybe a flat buff to morale regeneration based on how high the trending number is would help. The buff could also be a percentage of that number or multiplicative of it, depending on what would make the regeneration balanced. I don't know the code in this game. Also possible is a more long-term morale system we could call the happiness system, that keeps track of the little things that brings your character joy and through that applies buffs to morale regeneration. Maybe the more joy something brings to the character, the longer that instance of regeneration buff is held in their memory? Though broad categories could have to be applied, such as 'ate food' +14 (mock numbers) is supplanted by 'ate food' +30 for longer? Or they could be a small buff like 'ate toast-em' + 3% morale regeneration that together with other buffs like 'ate salmon' + 2% 'had beer + 4%' so the buffs work cumulatively. It could be reset within certain time frames too, like it clears memories of good things from 3 days ago.

I just want to say. I love melee builds, and I love two handed weapons. But the character I've had the most fun with though isn't the 16 strength hulks I want to make, because morale drain during combat is an issue for them. I want to add something to the game. I'm learning python at my own pace, and I hope that these suggestions will be at least food for thought. I want more variety, and more depth, and I will continue to support Bright Nights in the ways that I can. Thank you.

**Comment by MrOlaf** [comment]
Maybe the stamina system could be enhanced with Coward/Defensive/Normal/Aggressive/Berserk tactics settings - some would deplete stamina faster for bonuses similar to what is seen in ADOM's tactics settings? Just food for thought.

**Comment by scarf005** [comment]
closing as outdated. if necessary, please feel free to reopen them for further discussion."
